File: src/services/creative-types.ts
export interface CreativeTask {
  id: number;
  projectId: string | null;
  goalId: number | null;
  batchJobId: number | null;
  taskType: string;
  status: string;
  priority: number;
  payloadJson: string | null;
  resultJson: string | null;
  errorMessage: string | null;
  retryCount: number;
  maxRetries: number;
  parentTaskId: number | null;
  assetId: number | null;
  sequenceNo: number | null;
  createdAt: string;
  updatedAt: string;
  startedAt: string | null;
  finishedAt: string | null;
}

export interface CreateCreativeTaskInput {
  projectId?: string | null;
  goalId?: number | null;
  batchJobId?: number | null;
  taskType: string;
  status?: string | null;
  priority?: number | null;
  payloadJson?: string | null;
  maxRetries?: number | null;
  parentTaskId?: number | null;
  assetId?: number | null;
  sequenceNo?: number | null;
}

export interface ListCreativeTasksFilter {
  projectId?: string | null;
  status?: string | null;
  taskType?: string | null;
  goalId?: number | null;
  batchJobId?: number | null;
  limit?: number | null;
  offset?: number | null;
}

export interface UpdateCreativeTaskStatusInput {
  id: number;
  status: string;
  resultJson?: string | null;
  errorMessage?: string | null;
  assetId?: number | null;
  retryCountIncrement?: number | null;
}

export interface TaskEvent {
  id: number;
  taskId: number;
  eventType: string;
  message: string | null;
  payloadJson: string | null;
  createdAt: string;
}

export interface CreateTaskEventInput {
  taskId: number;
  eventType: string;
  message?: string | null;
  payloadJson?: string | null;
}

export interface CreativeAsset {
  id: number;
  projectId: string | null;
  assetType: string;
  title: string | null;
  content: string | null;
  filePath: string | null;
  thumbnailPath: string | null;
  metadataJson: string | null;
  status: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateCreativeAssetInput {
  projectId?: string | null;
  assetType: string;
  title?: string | null;
  content?: string | null;
  filePath?: string | null;
  thumbnailPath?: string | null;
  metadataJson?: string | null;
  status?: string | null;
}

export interface ListCreativeAssetsFilter {
  projectId?: string | null;
  assetType?: string | null;
  status?: string | null;
  limit?: number | null;
  offset?: number | null;
}

export interface CreativeAssetLink {
  id: number;
  sourceAssetId: number;
  targetAssetId: number;
  linkType: string;
  createdAt: string;
}

export interface CreateCreativeAssetLinkInput {
  sourceAssetId: number;
  targetAssetId: number;
  linkType: string;
}

export interface ListCreativeAssetLinksFilter {
  sourceAssetId?: number | null;
  targetAssetId?: number | null;
  linkType?: string | null;
  limit?: number | null;
  offset?: number | null;
}

export interface CreativeGoal {
  id: number;
  projectId: string | null;
  title: string;
  description: string | null;
  status: string;
  budgetJson: string | null;
  createdAt: string;
  updatedAt: string;
  startedAt: string | null;
  finishedAt: string | null;
  stoppedAt: string | null;
}

export interface CreateCreativeGoalInput {
  projectId?: string | null;
  title: string;
  description?: string | null;
  status?: string | null;
  budgetJson?: string | null;
}

export interface ListCreativeGoalsFilter {
  projectId?: string | null;
  status?: string | null;
  limit?: number | null;
  offset?: number | null;
}

export interface CreativeGoalRole {
  id: number;
  goalId: number;
  roleKey: string;
  taskType: string;
  description: string | null;
  taskCount: number;
  budgetJson: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface GoalAgentRoleSpecInput {
  roleKey: string;
  taskType: string;
  description?: string | null;
  taskCount?: number | null;
  budgetJson?: string | null;
}

export interface CreateGoalMultiAgentStubInput {
  projectId?: string | null;
  title: string;
  description?: string | null;
  budgetJson?: string | null;
  roleSpecs: GoalAgentRoleSpecInput[];
  mergeTaskType?: string | null;
}

export interface GoalMultiAgentStubResult {
  goal: CreativeGoal;
  roles: CreativeGoalRole[];
  tasks: CreativeTask[];
  mergeTask: CreativeTask | null;
}

export interface CreativeGoalStatusSnapshot {
  goal: CreativeGoal;
  roles: CreativeGoalRole[];
  tasks: CreativeTask[];
  totalTasks: number;
  queuedTasks: number;
  runningTasks: number;
  succeededTasks: number;
  failedTasks: number;
  cancelledTasks: number;
}

export interface CreativeBatchJob {
  id: number;
  projectId: string | null;
  name: string;
  batchType: string;
  status: string;
  totalCount: number;
  concurrency: number;
  maxRetries: number;
  promptTemplate: string | null;
  providerId: string | null;
  model: string | null;
  imageSize: string | null;
  budgetJson: string | null;
  createdAt: string;
  updatedAt: string;
  startedAt: string | null;
  finishedAt: string | null;
}

export interface CreativeBatchJobStats {
  totalTasks: number;
  draftTasks: number;
  queuedTasks: number;
  runningTasks: number;
  succeededTasks: number;
  failedTasks: number;
  cancelledTasks: number;
  cancellingTasks: number;
  paused: boolean;
  completionRatio: number;
}

export interface CreativeBatchJobSnapshot {
  job: CreativeBatchJob;
  stats: CreativeBatchJobStats;
}

export interface CreateBatchImageJobInput {
  projectId?: string | null;
  name: string;
  batchType?: string | null;
  totalCount?: number | null;
  concurrency?: number | null;
  maxRetries?: number | null;
  promptTemplate?: string | null;
  providerId?: string | null;
  model?: string | null;
  imageSize?: string | null;
  budgetJson?: string | null;
  providerConfig?: BatchPromptProviderConfigInput | null;
}

export interface BatchPromptProviderConfigInput {
  provider: string;
  displayName: string;
  baseUrl: string;
  apiKey: string;
  model: string;
  timeoutMs?: number | null;
  queueMode?: string | null;
  maxConcurrency?: number | null;
  queueKey?: string | null;
}

export interface ListCreativeBatchJobsFilter {
  projectId?: string | null;
  status?: string | null;
  batchType?: string | null;
  limit?: number | null;
  offset?: number | null;
}

export interface CreativeBatchJobEventPayload {
  batchJobId: number;
  projectId: string | null;
  batchType: string;
  status: string;
  totalTasks: number;
  queuedTasks: number;
  runningTasks: number;
  succeededTasks: number;
  failedTasks: number;
  cancelledTasks: number;
  createdAt: string;
  message: string | null;
}

export interface CreativeTaskEventPayload {
  taskId: number;
  projectId: string | null;
  status: string;
  message: string | null;
  createdAt: string;
}

export interface GenerateImagePromptWorkflowInput {
  projectId?: string | null;
  brief: string;
  style?: string | null;
  mood?: string | null;
  aspectRatio?: string | null;
}

export interface GenerateImagePromptWorkflowResult {
  task: CreativeTask;
  asset: CreativeAsset;
  events: TaskEvent[];
}

export interface ReviewAssetQualityStubInput {
  projectId?: string | null;
  sourceAssetId: number;
  sourceTaskId?: number | null;
  reviewKind?: string | null;
  contentHint?: string | null;
}

export interface ReviewResultPayload {
  pass: boolean;
  qualityScore: number;
  problems: string[];
  revisionInstruction: string;
  manualApprovalStatus: string;
  reviewKind: string;
  sourceAssetId: number;
  sourceTaskId: number | null;
}

export interface ReviewAssetQualityStubResult {
  task: CreativeTask;
  reviewAsset: CreativeAsset;
  reviseTask: CreativeTask | null;
  events: TaskEvent[];
}

File: src-tauri/src/infra/creative_types.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreativeTask {
    pub id: i64,
    pub project_id: Option<String>,
    pub goal_id: Option<i64>,
    pub batch_job_id: Option<i64>,
    pub task_type: String,
    pub status: String,
    pub priority: i64,
    pub payload_json: Option<String>,
    pub result_json: Option<String>,
    pub error_message: Option<String>,
    pub retry_count: i64,
    pub max_retries: i64,
    pub parent_task_id: Option<i64>,
    pub asset_id: Option<i64>,
    pub sequence_no: Option<i64>,
    pub created_at: String,
    pub updated_at: String,
    pub started_at: Option<String>,
    pub finished_at: Option<String>,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateCreativeTaskInput {
    pub project_id: Option<String>,
    pub goal_id: Option<i64>,
    pub batch_job_id: Option<i64>,
    pub task_type: String,
    pub status: Option<String>,
    pub priority: Option<i64>,
    pub payload_json: Option<String>,
    pub max_retries: Option<i64>,
    pub parent_task_id: Option<i64>,
    pub asset_id: Option<i64>,
    pub sequence_no: Option<i64>,
}

#[derive(Debug, Clone, Default, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ListCreativeTasksFilter {
    pub project_id: Option<String>,
    pub status: Option<String>,
    pub task_type: Option<String>,
    pub goal_id: Option<i64>,
    pub batch_job_id: Option<i64>,
    pub limit: Option<i64>,
    pub offset: Option<i64>,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UpdateCreativeTaskStatusInput {
    pub id: i64,
    pub status: String,
    pub result_json: Option<String>,
    pub error_message: Option<String>,
    pub asset_id: Option<i64>,
    pub retry_count_increment: Option<i64>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreativeBatchJob {
    pub id: i64,
    pub project_id: Option<String>,
    pub name: String,
    pub batch_type: String,
    pub status: String,
    pub total_count: i64,
    pub concurrency: i64,
    pub max_retries: i64,
    pub prompt_template: Option<String>,
    pub provider_id: Option<String>,
    pub model: Option<String>,
    pub image_size: Option<String>,
    pub budget_json: Option<String>,
    pub created_at: String,
    pub updated_at: String,
    pub started_at: Option<String>,
    pub finished_at: Option<String>,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateCreativeBatchJobInput {
    pub project_id: Option<String>,
    pub name: String,
    pub batch_type: String,
    pub status: Option<String>,
    pub total_count: Option<i64>,
    pub concurrency: Option<i64>,
    pub max_retries: Option<i64>,
    pub prompt_template: Option<String>,
    pub provider_id: Option<String>,
    pub model: Option<String>,
    pub image_size: Option<String>,
    pub budget_json: Option<String>,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UpdateCreativeBatchJobInput {
    pub id: i64,
    pub status: Option<String>,
    pub concurrency: Option<i64>,
    pub max_retries: Option<i64>,
    pub prompt_template: Option<String>,
    pub provider_id: Option<String>,
    pub model: Option<String>,
    pub image_size: Option<String>,
    pub budget_json: Option<String>,
    pub started_at: Option<String>,
    pub finished_at: Option<String>,
}

#[derive(Debug, Clone, Default, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ListCreativeBatchJobsFilter {
    pub project_id: Option<String>,
    pub status: Option<String>,
    pub batch_type: Option<String>,
    pub limit: Option<i64>,
    pub offset: Option<i64>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreativeBatchJobStats {
    pub total_tasks: i64,
    pub draft_tasks: i64,
    pub queued_tasks: i64,
    pub running_tasks: i64,
    pub succeeded_tasks: i64,
    pub failed_tasks: i64,
    pub cancelled_tasks: i64,
    pub cancelling_tasks: i64,
    pub paused: bool,
    pub completion_ratio: f64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreativeBatchJobSnapshot {
    pub job: CreativeBatchJob,
    pub stats: CreativeBatchJobStats,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ModelRun {
    pub id: i64,
    pub project_id: Option<String>,
    pub task_id: Option<i64>,
    pub asset_id: Option<i64>,
    pub provider_id: Option<String>,
    pub provider_type: Option<String>,
    pub model: Option<String>,
    pub request_type: String,
    pub status: String,
    pub duration_ms: Option<i64>,
    pub prompt_hash: Option<String>,
    pub prompt_version_id: Option<String>,
    pub input_token_count: Option<i64>,
    pub output_token_count: Option<i64>,
    pub cost_estimate: Option<f64>,
    pub error_code: Option<String>,
    pub error_message: Option<String>,
    pub metadata_json: Option<String>,
    pub created_at: String,
    pub finished_at: Option<String>,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateModelRunInput {
    pub project_id: Option<String>,
    pub task_id: Option<i64>,
    pub asset_id: Option<i64>,
    pub provider_id: Option<String>,
    pub provider_type: Option<String>,
    pub model: Option<String>,
    pub request_type: String,
    pub status: String,
    pub duration_ms: Option<i64>,
    pub prompt_hash: Option<String>,
    pub prompt_version_id: Option<String>,
    pub input_token_count: Option<i64>,
    pub output_token_count: Option<i64>,
    pub cost_estimate: Option<f64>,
    pub error_code: Option<String>,
    pub error_message: Option<String>,
    pub metadata_json: Option<String>,
    pub finished_at: Option<String>,
}

#[cfg(test)]
#[derive(Debug, Clone, Default, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ListModelRunsFilter {
    pub project_id: Option<String>,
    pub task_id: Option<i64>,
    pub asset_id: Option<i64>,
    pub request_type: Option<String>,
    pub status: Option<String>,
    pub limit: Option<i64>,
    pub offset: Option<i64>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct TaskEvent {
    pub id: i64,
    pub task_id: i64,
    pub event_type: String,
    pub message: Option<String>,
    pub payload_json: Option<String>,
    pub created_at: String,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateTaskEventInput {
    pub task_id: i64,
    pub event_type: String,
    pub message: Option<String>,
    pub payload_json: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreativeAsset {
    pub id: i64,
    pub project_id: Option<String>,
    pub asset_type: String,
    pub title: Option<String>,
    pub content: Option<String>,
    pub file_path: Option<String>,
    pub thumbnail_path: Option<String>,
    pub metadata_json: Option<String>,
    pub status: String,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreativeProject {
    pub id: String,
    pub title: String,
    pub description: Option<String>,
    pub status: String,
    pub settings_json: Option<String>,
    pub budget_json: Option<String>,
    pub created_at: String,
    pub updated_at: String,
    pub archived_at: Option<String>,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UpsertCreativeProjectInput {
    pub id: String,
    pub title: String,
    pub description: Option<String>,
    pub status: Option<String>,
    pub settings_json: Option<String>,
    pub budget_json: Option<String>,
    pub archived_at: Option<String>,
}

#[derive(Debug, Clone, Default, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ListCreativeProjectsFilter {
    pub status: Option<String>,
    pub limit: Option<i64>,
    pub offset: Option<i64>,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateCreativeAssetInput {
    pub project_id: Option<String>,
    pub asset_type: String,
    pub title: Option<String>,
    pub content: Option<String>,
    pub file_path: Option<String>,
    pub thumbnail_path: Option<String>,
    pub metadata_json: Option<String>,
    pub status: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreativeGoal {
    pub id: i64,
    pub project_id: Option<String>,
    pub title: String,
    pub description: Option<String>,
    pub status: String,
    pub budget_json: Option<String>,
    pub created_at: String,
    pub updated_at: String,
    pub started_at: Option<String>,
    pub finished_at: Option<String>,
    pub stopped_at: Option<String>,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateCreativeGoalInput {
    pub project_id: Option<String>,
    pub title: String,
    pub description: Option<String>,
    pub status: Option<String>,
    pub budget_json: Option<String>,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UpdateCreativeGoalStatusInput {
    pub id: i64,
    pub status: String,
    pub stopped_at: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreativeGoalRole {
    pub id: i64,
    pub goal_id: i64,
    pub role_key: String,
    pub task_type: String,
    pub description: Option<String>,
    pub task_count: i64,
    pub budget_json: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateCreativeGoalRoleInput {
    pub goal_id: i64,
    pub role_key: String,
    pub task_type: String,
    pub description: Option<String>,
    pub task_count: Option<i64>,
    pub budget_json: Option<String>,
}

#[derive(Debug, Clone, Default, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ListCreativeGoalsFilter {
    pub project_id: Option<String>,
    pub status: Option<String>,
    pub limit: Option<i64>,
    pub offset: Option<i64>,
}

#[derive(Debug, Clone, Default, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ListCreativeGoalRolesFilter {
    pub goal_id: Option<i64>,
    pub role_key: Option<String>,
    pub task_type: Option<String>,
    pub limit: Option<i64>,
    pub offset: Option<i64>,
}

#[derive(Debug, Clone, Default, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ListCreativeAssetsFilter {
    pub project_id: Option<String>,
    pub asset_type: Option<String>,
    pub status: Option<String>,
    pub limit: Option<i64>,
    pub offset: Option<i64>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AssetLink {
    pub id: i64,
    pub source_asset_id: i64,
    pub target_asset_id: i64,
    pub link_type: String,
    pub created_at: String,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateAssetLinkInput {
    pub source_asset_id: i64,
    pub target_asset_id: i64,
    pub link_type: String,
}

#[derive(Debug, Clone, Default, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ListAssetLinksFilter {
    pub source_asset_id: Option<i64>,
    pub target_asset_id: Option<i64>,
    pub link_type: Option<String>,
    pub limit: Option<i64>,
    pub offset: Option<i64>,
}
